upgrades/steps_27.go

```diff
@@ -64,3 +80,88 @@ func stateStepsFor27() []Step {
 		},
 	}
 }
+
+// stepsFor27 returns upgrade steps for Juju 2.7.
+func stepsFor27() []Step {
+	return []Step{
+		&upgradeStep{
+			description: "change owner of unit and machine logs to adm",
+			targets:     []Target{AllMachines},
+			run:         resetLogPermissions,
+		},
+	}
+}
+
+func setJujuFolderPermissionsToAdm(dir string) error {
+	wantedOwner, wantedGroup := paths.SyslogUserGroup()
+
+	err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return errors.Trace(err)
+		}
+		if info.IsDir() {
+			return nil
+		}
+		filePath := dir + string(os.PathSeparator) + info.Name()
+		if err := paths.SetOwnerGroupLog(filePath, wantedOwner, wantedGroup); err != nil {
+			return errors.Trace(err)
+		}
+		if err := os.Chmod(filePath, paths.LogfilePermission); err != nil {
+			return errors.Trace(err)
+		}
+		return nil
+	})
+	if err != nil {
+		return errors.Trace(err)
+	}
+	logger.Infof("Successfully changed permissions of dir %q", dir)
+	return nil
+}
+
+// This adds upgrade steps, we just rewrite the default values which are set before.
+// With this we can make sure that things are changed in one default place
+func resetLogPermissions(context Context) error {
+	tag := context.AgentConfig().Tag()
+	if tag.Kind() != names.MachineTagKind {
+		logger.Infof("skipping agent %q, not a machine", tag.String())
+		return nil
+	}
+	if context.AgentConfig().Value(agent.ProviderType) == k8sprovider.CAASProviderType {
+		logger.Infof("skipping agent %q, is CAAS controller", k8sprovider.CAASProviderType)
+		return nil
+	}
```

Comment:
I'm fairly sure we don't need this block, because the `Kind` of k8s agents are either Application or Controller. Since this is caught by the block above, we can simplify this.